lib/features/pokemon_details/pokemon_details_page.dart

```diff
@@ -1,16 +1,20 @@
+import 'package:dartx/dartx.dart';
 import 'package:flutter/material.dart';
 import 'package:pokedex/features/pokemon_details/widgets/about_tab_view.dart';
 import 'package:pokedex/features/pokemon_details/widgets/moves_tab_view.dart';
 import 'package:pokedex/features/pokemon_details/widgets/stats_tab_view.dart';
+import 'package:pokedex/model/dto/pokemon_dto.dart';
+import 'package:pokedex/utils/extensions.dart';
 import 'package:pokedex/utils/strings.dart' as str;
+import 'package:pokedex/utils/constant.dart' as k;
 
 class PokemonDetailsPage extends StatelessWidget {
   const PokemonDetailsPage({
     Key? key,
+    required this.pokemon,
   }) : super(key: key);
 
-  static const String route = 'details';
-  static const String routeName = 'details';
+  final PokemonDto? pokemon;
```

Comment:
will there be instances that you're gonna be entering this page without `pokemon` state? 🙂